src/controllers/fetch_email.js

```diff
@@ -315,10 +336,164 @@ export class FetchController extends BaseAPIController {
+        });
+
+        function findAttachmentParts(struct, attachments) {
+            attachments = attachments || [];
+            var len = struct.length;
+            for (var i = 0; i < len; ++i) {
+                if (Array.isArray(struct[i])) {
+                    findAttachmentParts(struct[i], attachments);
+                } else if (struct[i].disposition && ["INLINE", "ATTACHMENT"].indexOf(struct[i].disposition.type) > 0) {
+                    attachments.push(struct[i]);
+                }
+            }
+            return attachments;
+        }
+
+        function buildAttMessageFunction(attachment, uid, flag) {
+            var filename = attachment.disposition.params.filename;
+            var encoding = attachment.encoding;
+            var filepath = path.join(__dirname, "/./uploads/", filename);
+            return (msg, seqno) => {
+                var prefix = "(#" + seqno + ") ";
+                msg.on("body", (stream) => {
+                    var writeStream = fs.createWriteStream(filepath);
+                    if (encoding === "BASE64") {
+                        stream.pipe(base64.decode()).pipe(writeStream);
+                    } else {
+                        stream.pipe(writeStream);
+                    }
+                    fs.readFile(filepath, {
+                        encoding: "utf8"
+                    }, (error, data) => {
+                        var fileMetadata = {
+                            title: filename,
+                            mimeType: "text/plain/javascript/html/csv/application/pdf"
+                        };
+                        var media = {
+                            mimeType: "text/plain/javascript/html/csv/application/pdf",
+                            body: data
+                        };
+                        drive.files.insert({
+                            resource: fileMetadata,
+                            media: media,
+                            fields: "id"
+                        }, (err, file) => {
+                            if (!err) {
+                                var attachment_file = [{
+                                    name: attachment.disposition.params.filename,
+                                    link: "https://drive.google.com/file/d/" + file.id + "/view"
+                                }];
+                                resolve(attachment_file);
+                            } else {
+                                reject(err);
+                            }
+                        });
+                    });
+                    fs.unlink(filepath, () => {
+                        console.log("success");
+                    });
+                });
+                msg.once("end", () => {
+                    console.log(prefix + "Finished attachment %s", filename);
+                });
+            };
+        }
+        imap.once("error", (err) => {
+            reject(err);
+        });
+        imap.once("end", () => {
```

Comment:
does the imap connection close automatically? i don't see any code to close connection